clients/fides-js/src/fides-tcf.ts

```diff
@@ -80,28 +103,100 @@ declare global {
-  // specific consents. We rely on the cookie to fill those in. Therefore, we do nothing
-  // here, since the cookie is the source of truth, not the backend Experience.
-  return oldCookie;
+  const tcSavePrefs: TcfSavePreferences = {};
+  const enabledIds: EnabledIds = {
+    purposesConsent: [],
+    purposesLegint: [],
+    specialPurposes: [],
+    features: [],
+    specialFeatures: [],
+    vendorsConsent: [],
+    vendorsLegint: [],
+  };
+
+  TCF_KEY_MAP.forEach(({ experienceKey, cookieKey, enabledIdsKey }) => {
+    if (experienceKey) {
+      tcSavePrefs[cookieKey] = [];
+      experience[experienceKey]?.forEach((record) => {
+        const pref: UserConsentPreference = getInitialPreference(record);
+        // map experience to tcSavePrefs (same as cookie keys)
+        tcSavePrefs[cookieKey]?.push({
+          // @ts-ignore
+          id: record.id,
+          preference: pref,
+        });
+        // add to enabledIds only if user consent is True
+        if (transformUserPreferenceToBoolean(pref)) {
+          if (enabledIdsKey) {
+            enabledIds[enabledIdsKey].push(record.id.toString());
+          }
+        }
+      });
+    }
+  });
+
+  const tcString = await generateTcString({
+    experience,
+    tcStringPreferences: enabledIds,
+  });
+  const tcfConsent = transformTcfPreferencesToCookieKeys(tcSavePrefs);
+  return { ...oldCookie, tc_string: tcString, tcf_consent: tcfConsent };
 };
 
 /**
  * Initialize the global Fides object with the given configuration values
  */
 const init = async (config: FidesConfig) => {
   const cookie = getInitialCookie(config);
+  if (config.options.fidesString) {
+    // If a TC str is explicitly passed in, we override the associated cookie props, which are then used to
+    // override associated props in experience
+    debugLog(
+      config.options.debug,
+      "Explicit TC string detected. Proceeding to override all TCF preferences with given TC string"
+    );
+    cookie.tc_string = config.options.fidesString;
+    cookie.tcf_consent = transformTcStringToCookieKeys(
+      config.options.fidesString,
+      config.options.debug
+    );
+  } else if (
+    cookie.tcf_consent &&
+    !cookie.tc_string &&
+    isPrivacyExperience(config.experience) &&
+    experienceIsValid(config.experience, config.options)
+  ) {
+    // This state should not be hit, but just in case: if tc string is missing on cookie but we have tcf consent,
+    // we should generate tc string so that our CMP API accurately reflects user preference
+    cookie.tc_string = await generateTcStringFromCookieTcfConsent(
+      config.experience,
+      cookie.tcf_consent
+    );
+    debugLog(
+      config.options.debug,
+      "tc_string was missing from cookie, so it has been generated based on tcf_consent",
+      cookie.tc_string
+    );
+  }
   const initialFides = getInitialFides({ ...config, cookie });
   // Initialize the CMP API early so that listeners are established
   initializeCmpApi();
```

Comment:
@allisonking after rebasing with your work, `initializeCmpApi();` ended up down here, but I'm guessing we shouldn't put it above `getInitialFides()`?

Reply:
hm yes I think what you have is right